libs/core/src/scss/components/_grid.scss

```diff
@@ -0,0 +1,97 @@
+@use 'sass:map';
+
+/* stylelint-disable custom-property-pattern */
+$breakpoint-map: (
+  'xs': 0,
+  'sm': 632px,
+  'md': 768px,
+  'lg': 992px,
+  'xl': 1200px,
+);
+
+.kirby-grid {
+  --kirby-grid-columns: 12;
+  --kirby-grid-columnSpacing: 24px;
+  --kirby-grid-rowSpacing: 24px;
+
+  min-width: 0;
+  box-sizing: border-box;
+  display: flex;
+  flex-flow: row wrap;
+  margin: calc(var(--kirby-grid-rowSpacing) / -2) calc(var(--kirby-grid-columnSpacing) / -2);
+
+  &[spacing='1'],
+  &[data-spacing='1'] {
+    --kirby-grid-columnSpacing: 8px;
+    --kirby-grid-rowSpacing: 8px;
+  }
+
+  &[spacing='2'],
+  &[data-spacing='2'] {
+    --kirby-grid-columnSpacing: 16px;
+    --kirby-grid-rowSpacing: 16px;
+  }
+
+  &[spacing='3'],
+  &[data-spacing='3'] {
+    --kirby-grid-columnSpacing: 24px;
+    --kirby-grid-rowSpacing: 24px;
+  }
+
+  &[spacing='4'],
+  &[data-spacing='4'] {
+    --kirby-grid-columnSpacing: 32px;
+    --kirby-grid-rowSpacing: 32px;
+  }
+
+  &[spacing='5'],
+  &[data-spacing='5'] {
+    --kirby-grid-columnSpacing: 40px;
+    --kirby-grid-rowSpacing: 40px;
+  }
+}
+
+@mixin column-style($breakpoint, $size) {
+  &[#{$breakpoint}='#{$size}'],
+  &[data-#{$breakpoint}='#{$size}'] {
+    width: calc(100% * #{$size} / var(--kirby-grid-columns));
+    flex-grow: 0;
+    flex-basis: auto;
+  }
+}
+
+@mixin breakpoint-style($breakpoint) {
+  @media (min-width: map.get($breakpoint-map, $breakpoint)) {
+    @include column-style($breakpoint, 1);
+    @include column-style($breakpoint, 2);
+    @include column-style($breakpoint, 3);
+    @include column-style($breakpoint, 4);
+    @include column-style($breakpoint, 5);
+    @include column-style($breakpoint, 6);
+    @include column-style($breakpoint, 7);
+    @include column-style($breakpoint, 8);
+    @include column-style($breakpoint, 9);
+    @include column-style($breakpoint, 10);
+    @include column-style($breakpoint, 11);
+    @include column-style($breakpoint, 12);
+  }
+}
+
+.kirby-grid-item {
+  flex-grow: 1;
+  flex-basis: 0;
+  min-width: 0;
+  max-width: 100%;
+  box-sizing: border-box !important;
```

Comment:
Why is important needed here?

Reply:
I did this as the grid system relies on the box-sizing strategy. If anyone were to change this it would cause the entire grid to fail. I wanted to ensure this wouldn't happen even if someone would add something like that wouldn't happen if some projects were to include something that rewrites box-boxing with a "*" css selector :-) 

But I could remove it if you would prefer :-) ?

Reply:
Good call! Consider adding a comment explaining that.